### PR TITLE
bt-332: Adapt the sidebar for mobile layout for the whole project

### LINKED ISSUE #332: Adapt the sidebar for mobile layout for the whole project
The sidebar should hide on mobile and add a hamburger menu on the header

### PR DESCRIPTION
**Pull request by Monyster**
Closes #332 

width < 660 

image
image

### CODE REVIEW

**Review by JosuerBague — CHANGES_REQUESTED** [review]
Looks great and works great. 

I see that there is a dis-alignment between the burger menu and the rest of the onboarding content:
image

And perhaps this is out of scope for this ticket but changing padding in the following places will make it look better:

`frontend/src/bundles/talent-onboarding/pages/onboarding`:

```
@media screen and (width <=660px) {
    .career {
        padding-inline: 30px;
    }

    .header {
        padding-left: 30px;
    }
}

```

**Comment by Monyster** [comment]
> Looks great and works great.
> 
> I see that there is a dis-alignment between the burger menu and the rest of the onboarding content: ...
> 
> And perhaps this is out of scope for this ticket but changing padding in the following places will make it look better:
> 
> `frontend/src/bundles/talent-onboarding/pages/onboarding`:
> 
> ```
> @media screen and (width <=660px) {
>     .career {
>         padding-inline: 30px;
>     }
> 
>     .header {
>         padding-left: 30px;
>     }
> }
> ```
This task about the page layout and sidebar, so I did everything for that.
Paddings inside from another task. So maybe someone has already fixed it.
image

File: frontend/src/bundles/common/components/sidebar/styles.module.scss [hunk 1]
```diff
@@ -48,3 +53,125 @@ a {
...
+    position: relative;
+    box-sizing: content-box;
+    height: var(--size);
+    padding: 0;
+    color: inherit;
+    font: inherit;
+    background:
+        var(--g2) left var(--p, 0) top,
+        var(--g1) left calc(var(--size) / 10 + var(--p, 0px)) top,
+        var(--g2) right var(--p, 0) top,
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/sidebar/styles.module.scss]: you don't have --p variable here

File: frontend/src/bundles/common/components/sidebar/styles.module.scss [hunk 2]
```diff
@@ -48,3 +53,125 @@ a {
...
+        repeat-y content-box border-box;
+
+    position: relative;
+    box-sizing: content-box;
+    height: var(--size);
+    padding: 0;
+    color: inherit;
+    font: inherit;
+    background:
+        var(--g2) left var(--p, 0) top,
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/sidebar/styles.module.scss]: the same

File: frontend/src/bundles/common/components/sidebar/styles.module.scss [hunk 3]
```diff
@@ -48,3 +53,125 @@ a {
...
+    max-width: var(--sidebar-width);
+    height: var(--header-height);
+}
+
+.burgerButton {
+    --size: 20px;
+    --color: var(--text-primary-dark);
+    --g1: linear-gradient(var(--color) 20%, #00000000 0 80%, var(--color) 0)
+        no-repeat content-box border-box;
+    --g2: radial-gradient(
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/sidebar/styles.module.scss]: make normal naming for this variables

File: frontend/src/bundles/common/components/sidebar/styles.module.scss [hunk 4]
```diff
@@ -48,3 +53,125 @@ a {
...
+    background: var(--color);
+    border-radius: var(--size);
+    transition: transform 0.3s calc(0.3s - var(--s, 0.3s));
+    content: "";
+}
+
+.checked {
+    --color: #ffd231;
+    --p: calc(-1 * var(--size));
+    --s: 0s;
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/sidebar/styles.module.scss]: naming

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/sidebar/styles.module.scss (hunk 1 above)
Comment (Monyster) [reply frontend/src/bundles/common/components/sidebar/styles.module.scss]: image
--p  --> --position

File: frontend/src/bundles/common/components/sidebar/styles.module.scss [hunk 5]
```diff
@@ -41,3 +46,129 @@ a {
...
+    display: none;
+    width: var(--sidebar-width);
+    min-width: var(--sidebar-width);
+    max-width: var(--sidebar-width);
+    height: var(--header-height);
+}
+
+.burgerButton {
+    --size: 20px;
+    --color: var(--text-primary-dark);
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/sidebar/styles.module.scss:79]: looks like you can use our common variable

File: frontend/src/bundles/common/components/sidebar/styles.module.scss [hunk 6]
```diff
@@ -41,3 +46,129 @@ a {
...
+    --gradient1: linear-gradient(
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/sidebar/styles.module.scss]: we dont use such naming

File: frontend/src/bundles/common/components/sidebar/styles.module.scss [hunk 7]
```diff
@@ -41,3 +46,129 @@ a {
...
+            var(--color) 20%,
+            #00000000 0 80%,
+            var(--color) 0
+        )
+        no-repeat content-box border-box;
+    --gradient2: radial-gradient(
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/sidebar/styles.module.scss]: naming

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/sidebar/styles.module.scss (hunk 5 above)
Comment (Monyster) [reply frontend/src/bundles/common/components/sidebar/styles.module.scss:79]: As for the "--color", this variable is not static, it changes from var(--text-primary-dark) to #ffd231 after this button is clicked
I don't know how to make this animation without --color

**Comment by ptarho** [comment]
The logo is not centered, but the rest looks good
Screenshot_5

**Review by nikita-remeslov — APPROVED** [review]
Fix logo please

**Review by ptarho — APPROVED** [review]

**Review by anton-otroshchenko — APPROVED** [review]

**Pull request merged**